src/sources/GeoJSON.js

```diff
@@ -386,18 +391,22 @@ export default class GeoJSON extends Base {
             clipped: []
         };
         let holeIndex = 0;
+        let firstReverse = false;
         for (let i = 0; i < data.length; i++) {
-            for (let j = 0; j < data[i].length; j++) {
-                const point = this._computePointGeometry(data[i][j]);
-                polygon.flat.push(point.x, point.y);
+            let reverse = this._isReversed(data[i]);
+            if (i === 0) {
+                firstReverse = reverse;
+            } else if (firstReverse !== reverse) {
+                holeIndex += data[i - 1].length;
+                polygon.holes.push(holeIndex);
             }
-            if (!this._isClockWise(data[i])) {
-                if (i > 0) {
-                    holeIndex += data[i - 1].length;
-                    polygon.holes.push(holeIndex);
-                } else {
-                    throw new CartoValidationError('source', 'firstPolygonExternal');
-                }
+
+            let l = data[i].length;
```

Comment:
`l` => `dataLength` and use `const` instead of `let`

Reply:
Sure